backend/app/api/endpoints/adapter/models.py

```
@@ -530,6 +530,44 @@ def _test_embedding_connection(
+    elif provider_type == "custom":
+        """Test custom OpenAI-compatible embedding API connection."""
+        from langchain_openai import OpenAIEmbeddings
+
+        if not base_url:
+            return {
+                "success": False,
+                "message": "base_url is required for custom provider",
+            }
+
+        # Construct embeddings endpoint
+        embeddings_base_url = f"{base_url.rstrip('/').rstrip('/embeddings')}"
```

Ruff (0.14.13)

Using `.strip()` with multi-character strings is misleading

(B005)